### PR TITLE
Update ServiceExport status

### PR DESCRIPTION
**Pull request by aswinsuryan**
The service export status will be updated to ready when the
mcs CR is written in the broker.

Fixes: Fixes: submariner-io/lighthouse#164

Signed-off-by: Aswin Surayanarayanan <asuryana@redhat.com>

### CODE REVIEW

File: pkg/agent/controller/agent.go [hunk 1]
```diff
@@ -89,9 +99,43 @@ func (a *Controller) serviceExportToRemoteMcs(obj runtime.Object) runtime.Object
 			},
 		},
 	}
+	err = a.updateExportedServiceStatus(svcExport)
+	if err != nil {
+		klog.Errorf("Updating status failed %v", svcExport)
```
Comment (tpantelis) [inline pkg/agent/controller/agent.go:104]:
```suggestion
		klog.Errorf("Error updating status for %#v: %v", svcExport, err)
```

File: pkg/agent/controller/agent.go [hunk 2]
```diff
@@ -89,9 +99,43 @@ func (a *Controller) serviceExportToRemoteMcs(obj runtime.Object) runtime.Object
...
 }
 
+func (a *Controller) updateExportedServiceStatus(export *lighthousev2a1.ServiceExport) error {
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		toUpdate, err := a.lighthouseClient.LighthouseV2alpha1().ServiceExports(export.Namespace).Get(export.Name, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			klog.Infof("Export Service not found, hence ignoring %v", export)
+			return nil
+		} else if err != nil {
+			klog.Errorf("Error retrieving the object %v", export)
```
Comment (tpantelis) [inline pkg/agent/controller/agent.go:116]: Remove this as it's logged by the caller.

File: pkg/agent/controller/agent.go [hunk 3]
```diff
@@ -89,9 +99,43 @@ func (a *Controller) serviceExportToRemoteMcs(obj runtime.Object) runtime.Object
...
+			return err
+		}
+		msg := "Service written to the broker Successfully"
```
Comment (tpantelis) [inline pkg/agent/controller/agent.go:119]:
```suggestion
		msg := "Service was successfully synced to the broker"
```

File: test/e2e/framework/framework.go [hunk 4]
```diff
@@ -64,6 +64,22 @@ func (f *Framework) NewServiceExport(cluster framework.ClusterIndex, name string
...
+func (f *Framework) GetServiceExport(cluster framework.ClusterIndex, name string, namespace string) *v2alpha1.ServiceExport {
+	se := LighthouseClients[cluster].LighthouseV2alpha1().ServiceExports(namespace)
+	By(fmt.Sprintf("Retrieving serviceExport %s.%s on %q", name, namespace, framework.TestContext.ClusterIDs[cluster]))
+	serviceExport := framework.AwaitUntil("retrieve serviceExport", func() (interface{}, error) {
+		return se.Get(name, metav1.GetOptions{})
+
+	}, func(result interface{}) (bool, string, error) {
+		se := result.(*v2alpha1.ServiceExport)
+		if len(se.Status.Conditions) == 0 {
+			return false, fmt.Sprintf("expected Status to be non-empty"), nil
```
Comment (tpantelis) [inline test/e2e/framework/framework.go:76]:
```suggestion
			return false, "Status.Conditions is empty"), nil
```

File: test/e2e/framework/framework.go [hunk 5]
```diff
@@ -64,6 +64,22 @@ func (f *Framework) NewServiceExport(cluster framework.ClusterIndex, name string
 	return serviceExport
 }
 
+func (f *Framework) GetServiceExport(cluster framework.ClusterIndex, name string, namespace string) *v2alpha1.ServiceExport {
```
Comment (tpantelis) [inline test/e2e/framework/framework.go:67]:
```suggestion
func (f *Framework) AwaitServiceExportStatusCondition(cluster framework.ClusterIndex, name string, namespace string) *v2alpha1.ServiceExport {
```

File: pkg/agent/controller/agent.go [hunk 6]
```diff
@@ -89,9 +99,43 @@ func (a *Controller) serviceExportToRemoteMcs(obj runtime.Object) runtime.Object
 			},
 		},
 	}
+	err = a.updateExportedServiceStatus(svcExport)
```
Comment (tpantelis) [inline pkg/agent/controller/agent.go:102]: Of course the problem with updating it here is that it hasn't actually been synced yet and could fail - I guess this is where we need https://github.com/submariner-io/admiral/issues/66 😄  I guess we can move this call once that is done.

Comment (aswinsuryan) [reply pkg/agent/controller/agent.go:102]: yes that is right, we can change this one then.

File: pkg/agent/controller/agent.go [hunk 7]
```diff
@@ -89,9 +99,42 @@ func (a *Controller) serviceExportToRemoteMcs(obj runtime.Object) runtime.Object
...
+	if err != nil {
+		klog.Errorf("Error updating status for %#v: %v", svcExport, err)
+	}
 	return mcs
 }
 
+func (a *Controller) updateExportedServiceStatus(export *lighthousev2a1.ServiceExport) error {
```
[the remark this answers is not in the document]
Comment (aswinsuryan) [reply pkg/agent/controller/agent.go]: Changed to parameters.

**Pull request merged**